bin/test/main.ml

```diff
@@ -263,16 +263,15 @@ let with_non_det ~command ~output ~det non_deterministic = function
   | Some Label.Nd_output when not non_deterministic -> output ()
   | _ -> det ()
 
-let eval_prelude ?root top prelude prelude_str =
+let preludes ~prelude ~prelude_str =
   let aux to_lines p =
-    let env, f = Mdx.Prelude.env_and_file p in
-    let eval () = eval_raw ?root top ~line:0 (to_lines f) in
-    Mdx_top.in_env env eval
+    let env, file = Mdx.Prelude.env_and_file p in
+    (env, to_lines file)
   in
   match (prelude, prelude_str) with
-  | [], [] -> ()
-  | [], fs -> List.iter (aux (fun x -> [ x ])) fs
-  | fs, [] -> List.iter (aux Mdx.Util.File.read_lines) fs
+  | [], [] -> []
+  | [], fs -> List.map (aux (fun x -> [ x ])) fs
+  | fs, [] -> List.map (aux Mdx.Util.File.read_lines) fs
   | _ -> Fmt.failwith "only one of --prelude or --prelude-str shoud be used"
```

Comment:
Do you remember why this is forbidden? Couldn't we use a `[Prelude of string | Prelude_str of string] list` to take care of the evaluation order? (if this was the issue)

Reply:
I don't remember, I'll take a look but to be honest I think we'll just drop support for string preludes in the future.

Reply:
From a quick look it's always been like this. I don't think it's worth changing it!

Reply:
Alright thanks for looking at this